src/ORM/Rule/ExistsIn.php

```diff
@@ -129,4 +146,28 @@ protected function _fieldsAreNull($entity, $source)
         }
         return $nulls === count($this->_fields);
     }
+
+    /**
+     * Checks whether or not the give entity fields are null and map to schema NULL
+     * or are not null and map to schema NOT NULL.
+     *
+     * @param \Cake\Datasource\EntityInterface $entity The entity to check.
+     * @param \Cake\ORM\Table $source The table to use schema from.
+     * @return bool
+     */
+    protected function _checkPartialSchemaNulls($entity, $source)
+    {
+        $schema = $source->schema();
+        foreach ($this->_fields as $field) {
+            $isNullable = $schema->isNullable($field);
+            $value = $entity->get($field);
+            if (!$isNullable && $value === null) {
+                return false;
+            }
+            if ($isNullable && $value !== null) {
+                return false;
```

Comment:
Sorry for my bad explanation. Shouldn't the method `return true` in the following case?
```sql
CREATE TABLE t3 (
    x int NULL, -- Nullable
    y int NULL, -- Nullable
    FOREIGN KEY(x, y) REFERENCES t1(x, y)
);

-- Passing a not null value to a nullable column
INESRT INTO t3 VALUES (1, NULL);
```
When `$field` is `'x'`, `$isNullable` would be `true`, and `$value !== null` would be `true`. So I think it would `return false` wrongly.